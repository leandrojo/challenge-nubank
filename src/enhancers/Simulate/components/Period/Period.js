import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import {
  Box,
  Range,
} from 'components';

import Label from '../Label';

const Wrapper = styled(Box)`
  flex: none;
  flex-direction: column;
`;

const Meta = styled.span`
  color: ${p => p.theme.colors.primary};
  font-size: 1.2em;
  font-weight: 300;
  margin: 16px 0 12px;
`;

const Period = ({ css, onChange, value }) => {
  const handleChange = v => onChange(parseInt(v, 0));

  return (
    <Wrapper css={css}>
      <Label>For</Label>
      <Meta>{`${value} month${value === 1 ? '' : 's'}`}</Meta>
      <Range min={1} max={240} onChange={handleChange} value={value} />
    </Wrapper>
  );
};

Period.defaultProps = {
  css: {},
  onChange: () => {},
  value: 60,
};

Period.propTypes = {
  css: PropTypes.shape({}),
  onChange: PropTypes.func,
  value: PropTypes.number,
};

export default Period;
