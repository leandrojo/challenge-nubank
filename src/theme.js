const fontFamily = 'Rubik, sans-serif';

const theme = {
  backgroundColor: '#f5f5f5',
  colors: {
    primary: '#7f39b7',
    grayExtraDark: '#555',
    grayDark: '#767676',
    gray: '#999',
    grayLight: '#ccc',
    grayExtraLight: '#FAFAFA',
  },
  fontFamily,
  fontSize: '1em',
  typography: {
    heading: {
      color: '#252226',
    },
    paragraph: {
      color: '#716d73',
    },
  },
};

theme.components = {
  input: {
    css: {
      backgroundColor: 'transparent',
      fontFamily,
      fontSize: '1.4em',
      fontWeight: 300,
    },
  },
  pane: {
    backgroundColor: '#fff',
  },
  range: {
    emptyColor: theme.colors.grayLight,
  },
};

export default theme;
