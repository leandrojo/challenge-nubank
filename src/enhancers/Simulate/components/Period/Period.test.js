/* eslint-env jest */

import React, { useState } from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';

import theme from 'theme';

import Period from './Period';

describe('Simulate/Period', () => {
  it('with initial value', () => {
    const Component = () => {
      const [value, onChange] = useState(10);
      return (
        <ThemeProvider theme={theme}>
          <Period max={20} min={1} onChange={onChange} value={value} />
        </ThemeProvider>
      );
    };

    const expected = renderer.create(<Component />).toJSON();

    expect(expected).toMatchSnapshot();
  });
});
