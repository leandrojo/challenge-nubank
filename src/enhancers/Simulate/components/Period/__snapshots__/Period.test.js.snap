// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simulate/Period with initial value 1`] = `
<div
  className="sc-bwzfXH sc-VigVT jbIknr"
>
  <label
    className="sc-gqjmRU hJTcQd"
  >
    For
  </label>
  <span
    className="sc-jTzLTM nloSc"
  >
    10 months
  </span>
  <div
    className="sc-dnqmqq PpuwV"
  >
    <input
      className="sc-iwsKbI gUpUyg"
      max={240}
      min={1}
      name=""
      onChange={[Function]}
      type="range"
      value={10}
    />
    <div
      className="sc-gZMcBi dsEyqX"
      style={
        Object {
          "width": "3.7656903765690375%",
        }
      }
    />
  </div>
</div>
`;
